### PR TITLE
Navigation styling

### LINKED ISSUE #1666: Implement design updates to Navigation
- Blocked by #1658

### PR DESCRIPTION
**Pull request by rajatvijay**
Fixes #1666

Separate Issues
1. https://github.com/centerofci/mathesar/issues/2027
2. https://github.com/centerofci/mathesar/issues/2029

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by seancolsen — CHANGES_REQUESTED** [review]
This is looking good, and very close to merge-ready! 

Changes requested:

- On a URL like `http://localhost:8000/mathesar_tables/217/explorations/edit/5/` the breadcrumb structure doesn't match Figma. Can you fix this?

    | Figma | App |
    | -- | -- |
    | image | image |

- When you fix the conflicts that this PR now has, please make sure to test a record page like this:

    image

    Here the record summary has a `NULL` value inside it. I made changes in #2017 to stylize those values (hence the conflicts in your PR). Please make sure that the record summary looks good inside the navigation header now that it'll have a `NULL` value in it. If you run into problems you can ping me.

Notes

- In bb0650ad7b03960601d165fed198def9a55d49eb I fixed the issue with the logo. The SVG did actually have the white outline in it, but it was using an SVG `<mask>` element, which is [not well supported](https://caniuse.com/mdn-svg_elements_mask). Further, the actual SVG data was pretty messy with lots of superfluous path segments and nodes. I used Inkscape to clean it up, producing an SVG with only two (relatively simple) paths. I ended up with something _much_ smaller which I think is now suitable for inlining (hence the svelte file instead of svg file).

- I added an agenda item to our [front end team agenda](https://hackmd.io/ctEvo2tzQTyFV_38noSSEQ?edit) _"Refactoring ButtonMenuItem to use an onClick prop instead of a click event"_. This is to suggest an alternate approach to `ButtonMenuItem` that would lead to cleaner code inside the `AppHeader` you were just working on (because we wouldn't need to track the loading state in `AppHeader` anymore).

**Comment by rajatvijay** [comment]
> This is looking good, and very close to merge-ready!
> 
> Changes requested:
> 
>     * On a URL like `http://localhost:8000/mathesar_tables/217/explorations/edit/5/` the breadcrumb structure doesn't match Figma. Can you fix this?
>       
>       
>       
>       Figma
>       App
>       
>       
>       
>       
>       image
>       image

76932adb

>     * When you fix the conflicts that this PR now has, please make sure to test a record page like this:
>       image
>       Here the record summary has a `NULL` value inside it. I made changes in #2017 to stylize those values (hence the conflicts in your PR). Please make sure that the record summary looks good inside the navigation header now that it'll have a `NULL` value in it. If you run into problems you can ping me.

edda9c2ed1c781cb33d0804f75159fec8e138e81

> 
> Notes
> 
>     * In [bb0650a](https://github.com/centerofci/mathesar/commit/bb0650ad7b03960601d165fed198def9a55d49eb) I fixed the issue with the logo. The SVG did actually have the white outline in it, but it was using an SVG `<mask>` element, which is [not well supported](https://caniuse.com/mdn-svg_elements_mask). Further, the actual SVG data was pretty messy with lots of superfluous path segments and nodes. I used Inkscape to clean it up, producing an SVG with only two (relatively simple) paths. I ended up with something _much_ smaller which I think is now suitable for inlining (hence the svelte file instead of svg file).
> 
>     * I added an agenda item to our [front end team agenda](https://hackmd.io/ctEvo2tzQTyFV_38noSSEQ?edit) _"Refactoring ButtonMenuItem to use an onClick prop instead of a click event"_. This is to suggest an alternate approach to `ButtonMenuItem` that would lead to cleaner code inside the `AppHeader` you were just working on (because we wouldn't need to track the loading state in `AppHeader` anymore).

File: mathesar_ui/src/components/breadcrumb/BreadcrumbSelector.svelte [hunk 1]
```diff
@@ -25,17 +31,20 @@
...
     }}
     aria-label={triggerLabel}
     title={triggerLabel}
+    appearance="ghost"
+    bind:element={triggerElement}
+    class="padding-zero"
   >
-    <BreadcrumbSeparatorIcon />
-  </button>
+    <div class="trigger">
```
[the remark this answers is not in the document]
Comment (rajatvijay) [reply mathesar_ui/src/components/breadcrumb/BreadcrumbSelector.svelte]: c358a4c4

File: mathesar_ui/src/components/breadcrumb/BreadcrumbRecordSelector.svelte [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const recordSelector = getRecordSelectorFromContext();
+  export let table: TableEntry;
+
+  function handleClick() {
+    recordSelector.navigateToRecordPage({ tableId: table.id });
+  }
+</script>
+
+<Button on:click={handleClick} appearance="ghost" class="padding-zero">
+  <div class="trigger">
```
[the remark this answers is not in the document]
Comment (rajatvijay) [reply mathesar_ui/src/components/breadcrumb/BreadcrumbRecordSelector.svelte]: c358a4c4

**Review by seancolsen — APPROVED** [review]
Looks good. There is a small problem with `NULL` values within the record summary. I opened #2055 to handle it later.

**Pull request merged**